doc/design/GeneratedClientDesign.md

```diff
@@ -0,0 +1,133 @@
+# Generated Client Design
+
+## Introduction
+
+This document tracks design rules and decisions for .NET clients generated by the DPG generator.  It may later be extended to include other generator designs.
+
+<!--
+## Clients
+
+TBD
+
+## Client Protocol Operations
+
+TBD
+
+## Client Convenience Operations
+-->
+
+## Model Generation
+
+Models are defined as partial classes with one or more constructors, properties, and methods that read from/write to wire formats.
+
+### Model Namespace
+
+Generated models will be defined in a `.Models` namespace.
+
+### Type mappings from Cadl to .NET
+
+The following table shows a mapping from Cadl built-in types to the corresponding types used in .NET models.
+
+Cadl Type | .NET Type | OpenAPI Type | GitHub Issue | Notes
+------------------- | -------- | -- | -- | -------------
+string | string | string | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) | [PrimitivePropertyModel.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.cs) [PrimitivePropertyModel.Serialization.cs](https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-model-primitive-properties/sdk/template/Azure.Template/src/Generated/Models/PrimitivePropertyModel.Serialization.cs)
+bytes | bytes[] | type: string, format: byte | [autorest.csharp #2337](https://github.com/Azure/autorest.csharp/issues/2337) |
```

Comment:
Should it be Memory, ReadOnlyMemory\<T\>?

Need exact size byte array, may not know the exact size.

ReadOnlyMemory, allows slicing down.

Issue 2:
byte array is mutable.
ReadOnlyMemory doesn't allow changes.

Handle differently based on input/output/roundtrip/parameter.

Negatives of ReadOnlyMemory (need to know how used to decide):
- If BCL APIs want byte[], this is a negative.
- In apps, use ReadOnlyMemory
- If in a library, .NET Standard, limitations on how consumed.

When we manually design the APIs, use BinaryData.
Decision: Default to BinaryData, may find cases where this doesn't work.  We'll learn from these going forward.

Reply:
Question: extra work for serialization/deserialization logic - @AlexanderSher  to consider and make a recommendation.